src/app-layer-ssl.c

```diff
@@ -818,6 +819,56 @@ static inline int TLSDecodeHSHelloExtensionSni(SSLState *ssl_state,
     return -1;
 }
 
+static inline int TLSDecodeHSHelloExtensionSupportedVersions(SSLState *ssl_state,
+                                             const uint8_t * const initial_input,
+                                             const uint32_t input_len)
+{
+    uint8_t *input = (uint8_t *)initial_input;
+
+    if (ssl_state->current_flags & SSL_AL_FLAG_STATE_CLIENT_HELLO) {
+        if (!(HAS_SPACE(1)))
+            goto invalid_length;
+
+        uint8_t supported_ver_len = *input;
+        input += 1;
+
+        if (!(HAS_SPACE(supported_ver_len)))
+            goto invalid_length;
+
+        /* Use the first (and prefered) version as client version */
+        ssl_state->curr_connp->version = *input << 8 | *(input + 1);
```

Comment:
should we record all advertised versions here for logging and also perhaps validate that the top one is in fact the best version?

Reply:
The top one might not be the best TLS version, but it is the one that the client wants most to use.

Would it make sense to do the work of storing all advertised versions for logging purposes in another PR? I mean, since we don't store the client version now.

Reply:
Yeah that sounds good.